src/amazonS3.ts

```diff
@@ -133,17 +87,60 @@ export class AmazonS3 {
       .promise();
     return objects.Contents ?? [];
   }
+}
 
-  private resolveCredentials(): { credentials: CredentialsOptions } | Record<string, string> {
-    const credentials: { credentials: CredentialsOptions } | Record<string, string> = {};
-    if (this.options.credentials) {
-      return { credentials: this.options.credentials };
-    }
-    if (process.env.AWS_ACCESS_KEY_ID && process.env.AWS_SECRET_ACCESS_KEY) {
-      return {
-        credentials: { accessKeyId: process.env.AWS_ACCESS_KEY_ID, secretAccessKey: process.env.AWS_SECRET_ACCESS_KEY },
-      };
-    }
-    return credentials;
+export const download = async (url: string, location: string, silent = false): Promise<void> => {
+  const downloadStream = got.stream(url);
+  const fileWriterStream = fs.createWriteStream(location);
+  return new Promise((resolve) => {
+    downloadStream.on('error', (error) => {
+      if (!silent) ux.error(`Download failed: ${error.message}`);
+    });
+
+    fileWriterStream
+      .on('error', (error) => {
+        if (!silent) ux.action.stop('Failed');
+        if (!silent) ux.error(`Could not write file to system: ${error.message}`);
+      })
+      .on('finish', () => {
+        if (!silent) ux.action.stop();
+        resolve();
+      });
+    if (!silent) ux.action.start(`Downloading ${chalk.cyan(url)}`);
+    downloadStream.pipe(fileWriterStream);
+  });
+};
+
+const getFileAtUrl = async (url: string): Promise<string> => {
+  const availability = await fileIsAvailable(url);
+  if (availability.available) {
+    const filename = path.join(os.tmpdir(), `file${Math.random()}`);
+    await download(url, filename, true);
+    return filename;
+  } else {
+    throw new SfError(`File at url: ${url} does not exist`);
   }
-}
+};
+
+const resolveCredentials = (
+  credentialOptions?: CredentialsOptions
+): { credentials: CredentialsOptions } | Record<string, string> => {
+  if (credentialOptions) {
+    return { credentials: credentialOptions };
+  }
+  return process.env.AWS_ACCESS_KEY_ID && process.env.AWS_SECRET_ACCESS_KEY
+    ? {
+        credentials: { accessKeyId: process.env.AWS_ACCESS_KEY_ID, secretAccessKey: process.env.AWS_SECRET_ACCESS_KEY },
+      }
+    : {};
+};
+
+const fileIsAvailable = async (url: string): Promise<ServiceAvailability> => {
+  const { statusCode } = await got.get(url, { throwHttpErrors: false });
```

Comment:
since this only checks if a file is available let's make it do a `HEAD` request.
https://developer.mozilla.org/en-US/docs/web/http/methods/head

Reply:
😄 https://github.com/vercel/next.js/pull/14994

Reply:
good idea